### PR TITLE
Separate bundle and remove deprecated fields

### PR DESCRIPTION
**Pull request by lionel-bijaoui**
Separation into 2 bundles: "core" and "optionnal" #112 
removing deprecated fields #120

### CODE REVIEW

**Comment by lionel-bijaoui** [comment]
I will correct the merge problem beforehand

File: package.json [hunk 1]
```diff
@@ -6,7 +6,8 @@
   "main": "dist/vue-form-generator.js",
   "scripts": {
     "prebuild": "npm run test",
-    "build": "webpack --progress --config webpack.build.config.js",
+    "build": "webpack --progress --config webpack.build.config.js --define process.env.FULL_BUNDLE=true",
+    "buildcore": "webpack --progress --config webpack.build.config.js --define process.env.FULL_BUNDLE=false --output-filename=vue-form-generator-core.js",
     "dev": "webpack-dev-server --config webpack.dev.config.js --inline --hot --content-base dev/",
```
Comment (icebob) [inline package.json]: Please create 3 build script.
```
"build:core": "...."
"build:full": "...."
"build": "npm run build:core && npm run build:full"
```

Comment (lionel-bijaoui) [inline package.json]: ok I will add it

File: src/formGenerator.vue [hunk 2]
```diff
@@ -22,13 +22,25 @@ div
...
+	let coreFields = require.context("./fields/core", false, /^\.\/field([\w-_]+)\.vue$/);
+				
+	each(coreFields.keys(), (key) => {
 		let compName = key.replace(/^\.\//, "").replace(/\.vue/, "");
-		fieldComponents[compName] = Fields(key);
+		fieldComponents[compName] = coreFields(key);
 	});
 
+	if (process.env.FULL_BUNDLE) {  // eslint-disable-line
+		let Fields = require.context("./fields/optional", false, /^\.\/field([\w-_]+)\.vue$/);
```
Comment (cristijora) [inline src/formGenerator.vue]: Maybe name this `optionalFields` so it's more clear what it does.

Comment (lionel-bijaoui) [inline src/formGenerator.vue]: Yes, I wasn't sure. I will change it too.

Comment (icebob) [inline src/formGenerator.vue]: I don't agree with `optionalFields`, because in the future maybe we will add other stuffs to the `full` bundle.

**Comment by lionel-bijaoui** [comment]
Sorry, a bit of confusion on my part. I was not seeing the latest commit.
I had to skip some test. Switching to fecha and removing deprecated fields

**Pull request merged**